website_sale_adj/tests/test_website_sale_adj.py

```diff
@@ -74,3 +74,62 @@ def test_03_create_product(self):
         self.assertEqual(product_order_list[0], product_3)
         self.assertEqual(product_order_list[1], product_2)
         self.assertEqual(product_order_list[2], product_1)
+
+    def test_04_compute_order_line_date(self):
+        """
+            This tests check the earliest
+            website_order_line's creation date
+            """
+        self.product_uom = self.env.ref("product.product_uom_unit").id
+
+        # Create Test Product
+        self.product_01 = self.env["product.product"].create(
+            {"name": "Test Product 1", "type": "product"}
+        )
+        self.product_02 = self.env["product.product"].create(
+            {"name": "Test Product 2", "type": "product"}
+        )
+
+        # Create Test Partner
+        self.partner_01 = self.env["res.partner"].create({"name": "Test Partner 1"})
+
+        # Create sale order
+        self.sale_order = self.env["sale.order"].create(
+            {"partner_id": self.partner_01.id}
+        )
+
+        # Create Sale Order line
+        line_01 = self.env["sale.order.line"].create(
+            {
+                "product_id": self.product_01.id,
+                "price_unit": 100.00,
+                "product_uom": self.product_uom,
+                "product_uom_qty": 10.0,
+                "order_id": self.sale_order.id,
+            }
+        )
+        line_02 = self.env["sale.order.line"].create(
+            {
+                "product_id": self.product_02.id,
+                "price_unit": 200.00,
+                "product_uom": self.product_uom,
+                "product_uom_qty": 20.0,
+                "order_id": self.sale_order.id,
+            }
+        )
+
+        # Assign website order line from sale order line
+        self.sale_order.write(
+            {"website_order_line": [(6, 0, [line_01.id, line_02.id])]}
+        )
+
+        # Compute the `_compute_order_line_date`
+        self.sale_order._compute_order_line_date()
+
+        # Compare the `order_line_date` with sale order line create_date.
+        self.assertEqual(
+            self.sale_order.order_line_date,
+            line_01.create_date,
```

Comment:
We should first manipulate with `create_date` of line_01 (e.g. set it to `2020-01-01 00:00:00`), and then use the value for checking here.

Reply:
Seems not working if I add the value for create_date 
Ex

```
line_01 = self.env["sale.order.line"].create(
            {
                "product_id": self.product_01.id,
                "price_unit": 100.00,
                "create_date": '2020-01-01 00:00:00',
                "product_uom": self.product_uom,
                "product_uom_qty": 10.0,
                "order_id": self.sale_order.id,
            }
        )

```

Another way I have tried to update `create_date` 
```
        self.env.cr.execute('UPDATE sale_order_line SET create_date = %(date)s WHERE id = %(lind_id)s',
            {'date': '2020-01-01 00:00:00',
             'lind_id': line_01.id})
```

Reply:
Hi @TimLai125 I have tried to sleep function here for the delay to create order lines, but seems it's not working, also tried to threading with timer, but it also not working. I think for this module we need to be tested functionally.

Reply:
> Another way I have tried to update create_date

@YogeshMahera-SerpentCS I can understand how ORM wouldn't accept manipulation with create_date, but I do not understand how it does not work with direct SQL. Can you please double-check and provide more info.

Reply:
Hi @yostashiro 

Yes, I again checked still not updating the using SQL. 
Even it also not working on enterprise where there is the same example available. 
For more information, you may check-in helpdesk module v11 helpdesk-->tests-->test_helpdesk_sla.py-->Line 48

Reply:
@YogeshMahera-SerpentCS When you say "not working":
- Provide info on what exact error you get
- Provide your opinion/reasoning on what causes that error
- Provide info/reference that backs up your opinion/reasoning

My guess is that the direct SQL does not work because no records are committed to the database during the test. Can you please verify this point.

I think there is a way to manipulate with the create_date using something like https://pypi.org/project/freezegun/, but it is probably an overkill for this test. So, I'm OK with a separate functional test.

Reply:
Hi @TimLai125 Can we close this PR?